src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,146 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final int FIRST_PARAMETER_INDEX = 1;
+    private static final int SECOND_PARAMETER_INDEX = 2;
+    private static final int THIRD_PARAMETER_INDEX = 3;
+    private static final int FIRST_COLUMN_INDEX = 1;
+    private static final int SECOND_COLUMN_INDEX = 2;
+    private static final int THIRD_COLUMN_INDEX = 3;
+    private static final String SQL_CREATE_REQUEST = "INSERT INTO "
+            + "books (title, price) VALUES (?, ?)";
+    private static final String SQL_UPDATE_REQUEST = "UPDATE books"
+            + " SET title = ?, price = ? WHERE id = ?";
+    private static final String SQL_DELETE_REQUEST = "DELETE FROM"
+            + " books WHERE id = ?";
+    private static final String SQL_OUTPUT_REQUEST = "SELECT * "
+            + "FROM books";
+    private static final String SQL_BY_ID_REQUEST = "SELECT * "
+            + "FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                           .prepareStatement(SQL_CREATE_REQUEST, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(FIRST_PARAMETER_INDEX, book.getTitle());
+            preparedStatement.setBigDecimal(SECOND_PARAMETER_INDEX, book.getPrice());
+            int affectedRows = preparedStatement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to "
+                        + "inserted at least one row, but inserted 0 rows.");
+            }
+
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(FIRST_COLUMN_INDEX, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to establish connection"
+                    + " with the database", e);
+        }
+        return book;
```

Comment:
Use `PreparedStatement` over `Statement`, even for a static query with no parameters in `findAll()` method. It's the best practice, and it's slightly faster. Also, you're already using `PreparedStatement` so there's no need for `Statement.RETURN_GENERATED_KEYS` in the `prepareStatement` call in the `create` method.